core/src/test/java/kafka/admin/UserScramCredentialsCommandTest.java

```diff
@@ -0,0 +1,178 @@
+        String user1 = "user1";
+        // create and describe a credential
+        ConfigCommandResult result = runConfigCommandViaBroker("--user", user1, "--alter", "--add-config", "SCRAM-SHA-256=[iterations=4096,password=foo-secret]");
+        String alterConfigsUser1Out = "Completed updating config for user " + user1 + ".\n";
+        assertEquals(alterConfigsUser1Out, result.stdout);
+        String scramCredentialConfigsUser1Out = "SCRAM credential configs for user-principal '" + user1 + "' are SCRAM-SHA-256=iterations=4096\n";
+        TestUtils.waitForCondition(
+            () -> Objects.equals(runConfigCommandViaBroker("--user", user1, "--describe").stdout, scramCredentialConfigsUser1Out),
+            () -> "Failed to describe SCRAM credential change '" + user1 + "'");
+        // create a user quota and describe the user again
+        result = runConfigCommandViaBroker("--user", user1, "--alter", "--add-config", "consumer_byte_rate=20000");
+        assertEquals(alterConfigsUser1Out, result.stdout);
+        String quotaConfigsUser1Out = "Quota configs for user-principal '" + user1 + "' are consumer_byte_rate=20000.0\n";
+        TestUtils.waitForCondition(
+            () -> Objects.equals(runConfigCommandViaBroker("--user", user1, "--describe").stdout, quotaConfigsUser1Out + scramCredentialConfigsUser1Out),
+            () -> "Failed to describe Quota change for '" + user1 + "'");
+
+        // now do the same thing for user2
+        String user2 = "user2";
+        // create and describe a credential
+        result = runConfigCommandViaBroker("--user", user2, "--alter", "--add-config", "SCRAM-SHA-256=[iterations=4096,password=foo-secret]");
+        String alterConfigsUser2Out = "Completed updating config for user " + user2 + ".\n";
+        assertEquals(alterConfigsUser2Out, result.stdout);
+        String scramCredentialConfigsUser2Out = "SCRAM credential configs for user-principal '" + user2 + "' are SCRAM-SHA-256=iterations=4096\n";
+        TestUtils.waitForCondition(
+            () -> Objects.equals(runConfigCommandViaBroker("--user", user2, "--describe").stdout, scramCredentialConfigsUser2Out),
+            () -> "Failed to describe SCRAM credential change '" + user2 + "'");
+        // create a user quota and describe the user again
+        result = runConfigCommandViaBroker("--user", user2, "--alter", "--add-config", "consumer_byte_rate=20000");
+        assertEquals(alterConfigsUser2Out, result.stdout);
+        String quotaConfigsUser2Out = "Quota configs for user-principal '" + user2 + "' are consumer_byte_rate=20000.0\n";
+        TestUtils.waitForCondition(
+            () -> Objects.equals(runConfigCommandViaBroker("--user", user2, "--describe").stdout, quotaConfigsUser2Out + scramCredentialConfigsUser2Out),
+            () -> "Failed to describe Quota change for '" + user2 + "'");
+
+        // describe both
+        result = runConfigCommandViaBroker("--entity-type", "users", "--describe");
```

Comment:
Could you separate this test case?

Reply:
I extract some common code into methods and this made tests much more readable in my opinion.
Can you, please, take a look and tell me - if we still need to extract this to separate case?